chi.go

```diff
@@ -89,6 +95,21 @@ type Routes interface {
 	Middlewares() Middlewares
 }
 
-// Middlewares type is a slice of standard middleware handlers with methods
+// Middleware type represents a middleware handler signature.
+type Middleware func(http.Handler) http.Handler
+
+// Middlewares type is a slice of middleware handlers with methods
 // to compose middleware chains and http.Handler's.
 type Middlewares []func(http.Handler) http.Handler
+
+// MiddlewareFunc type is an adapter to allow the use of ordinary
+// functions as middleware handlers. If f is a function with the
+// appropriate signature, MiddlewareFunc(f) is a Handler that calls f.
+type MiddlewareFunc func(w http.ResponseWriter, r *http.Request, next http.HandlerFunc)
```

Comment:
Note how similar this is to `http.HandlerFunc`:
https://github.com/golang/go/blob/b20c055230a20663c75e3099f672c15c39d46b9e/src/net/http/server.go#L1893-L1902